src/App.tsx

```diff
@@ -15,36 +15,98 @@
   'Garlic',
 ];
 
+enum SortType {
+  None = 'none',
+  Alphabetically = 'alphabetically',
+  ByLength = 'length',
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+  const [sortMethod, setSortMethod] = useState<SortType>(SortType.None);
+
+  const sortGoods = (method: SortType, reverse: boolean): string[] => {
+    const sortedGoods = [...goodsFromServer];
+
+    if (method === SortType.Alphabetically) {
+      sortedGoods.sort();
+    } else if (method === SortType.ByLength) {
+      sortedGoods.sort((a, b) => a.length - b.length);
+    }
+
+    if (reverse) {
+      sortedGoods.reverse();
+    }
+
+    return sortedGoods;
+  };
+
+  const reverseHandle = () => {
+    setIsReversed(prev => !prev);
+  };
+
+  const sortAlphabeticallyHandle = () => {
+    setSortMethod(SortType.Alphabetically);
+  };
```

Comment:
[CODE STYLE] - Function's name should start with a verb. Instead of `sortAlphabeticallyHandle`, consider renaming it to `handleSortAlphabetically`.